### PR TITLE
Codepipeline

### PR DESCRIPTION
**Pull request by brianterry**
Fixes issue #48

### CODE REVIEW

File: events/codepipeline_job.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// CodePipelineConfiguration represents a configuration for an Action Configuration
+type CodePipelineConfiguration struct {
+	FunctionName   string `json:"FunctionName"`
+	UserParameters string `json:"UserParameters"`
+}
+
+// CodePipelineInputArtifact represents an input artifact
+type CodePipelineInputArtifact struct {
+	Location CodePipelineInputLocation `json:"location"`
+	Revision string                    `json:"revision"`
```
Comment (bmoffatt) [inline events/codepipeline_job.go]: regarding your comment on #48 
> In the unmarshal and marshal event, go converts this to " "

You can make the documented sample from https://docs.aws.amazon.com/codepipeline/latest/userguide/actions-invoke-lambda-function.html#actions-invoke-lambda-function-json-event-example work if you change this field's type to be pointer. `Revision *string`

Comment (brianterry) [reply events/codepipeline_job.go]: Changing `Revision string`  to `Revision *string` still results in Revision being converted to   `revision":" " ` if `"revision": null,` is given in the test doc. This causes tests to fail if using https://docs.aws.amazon.com/codepipeline/latest/userguide/actions-invoke-lambda-function.html#actions-invoke-lambda-function-json-event-example as a test doc.  Should I just replace `"revision": null,` with `"revision": "foo"` in the test doc?

Comment (bmoffatt) [reply events/codepipeline_job.go]: From https://play.golang.org/p/azCGae4c6jL, the type change ought to work.

**Comment by bmoffatt** [comment]
Make the `string` -> `*string` change and updated the test data. Thanks for the contribution!

**Comment by brianterry** [comment]
Woot!

**Pull request merged**